### DISCUSSION TITLE
Some questions about graphsage

### DISCUSSION THREAD

**Discussion by scottshufe** [pyg-team/pytorch_geometric#4115] [Q&A] [answered]
Hi, I would like to ask some questions about graphsage (or SAGEConv):
1. Is NeighborSampler necessary for a graphsage model?
2. Could you please provide me with some graphsage examples for the graph classification task?

(By the way, it often takes me a lot of time to seek the example codes I want. It would be great if your team could provide an index for the example codes in _README_ file in the example directory)

Thanks in advance :)

**Comment by rusty1s** [comment, marked as answer]
Thanks for your interest.

1. No, we disentangle the model (`SAGEConv`) from the scalability technique (`NeighborSampler`). That way, you can use `SAEGConv` for full-batch training, and can use any other PyG GNN model for neighbor sampling as well.
2. [Here](https://github.com/pyg-team/pytorch_geometric/blob/master/benchmark/kernel/graph_sage.py#L8-L37) is an example `GraphSAGE` model for graph classification.

**Reply by scottshufe** [reply]
Wow, thank you so much!

**Comment by rusty1s** [comment]
> (By the way, it often takes me a lot of time to seek the example codes I want. It would be great if your team could provide an index for the example codes in README file in the example directory)

I totally understand that. This is a great idea. Let me work on that.

**Reply by scottshufe** [reply]
:wink::wink:
